src/passes/CoalesceLocals.cpp

```diff
@@ -440,7 +440,7 @@ void CoalesceLocals::applyIndices(std::vector<Index>& indices, Expression* root)
   auto numParams = getFunction()->getNumParams();
   Index newNumLocals = 0;
   for (auto index : indices) {
-    newNumLocals = std::max(newNumLocals, index + 1);
+    newNumLocals = std::max((Index::value_type)newNumLocals, index + 1);
```

Comment:
(Also this only is needed when mixing Index with other types. There are already other uses of `std::max` or min on `Index` in the codebase and they are fine as-is).